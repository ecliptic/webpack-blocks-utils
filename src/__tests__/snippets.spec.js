// @flow
import {
  setTarget,
  setNode,
  setExternals,
  // envVar,
  noParse,
  resolveModules,
} from '../snippets'

describe('snippets', () => {
  test('setTarget()', () => {
    const block = setTarget('electron-renderer')
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    const result = block(context, helpers)({})
    expect(result).toMatchSnapshot()
  })

  test('setNode()', () => {
    const block = setNode({console: false})
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    const result = block(context, helpers)({})
    expect(result).toMatchSnapshot()
  })

  test('setExternals()', () => {
    const block = setExternals(['debug'])
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    const result = block(context, helpers)({})
    expect(result).toMatchSnapshot()
  })

  test('envVar()') // TODO

  test('noParse()', () => {
    const block = noParse(/node_modules/)
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    const result = block(context, helpers)({})
    expect(result).toMatchSnapshot()
  })

  test('resolveModules()', () => {
    const block = resolveModules(['node_modules', 'src'])
    const context = {}
    const helpers = {
      merge: configSnippet => prevConfig => configSnippet,
    }

    const result = block(context, helpers)({})
    expect(result).toMatchSnapshot()
  })
})
