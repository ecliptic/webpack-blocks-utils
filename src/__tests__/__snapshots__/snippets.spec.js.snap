// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippets noParse() 1`] = `
Object {
  "module": Object {
    "noParse": /node_modules/,
  },
}
`;

exports[`snippets resolveModules() 1`] = `
Object {
  "resolve": Object {
    "modules": Array [
      "node_modules",
      "src",
    ],
  },
}
`;

exports[`snippets setExternals() 1`] = `
Object {
  "externals": Array [
    "debug",
  ],
}
`;

exports[`snippets setNode() 1`] = `
Object {
  "node": Object {
    "console": false,
  },
}
`;

exports[`snippets setTarget() 1`] = `
Object {
  "target": "electron-renderer",
}
`;
